pkg/authorization/reaper/cluster_role.go

```diff
@@ -0,0 +1,52 @@
+package reaper
+
+import (
+	"time"
+
+	"github.com/golang/glog"
+	kapi "k8s.io/kubernetes/pkg/api"
+	kerrors "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/kubectl"
+	"k8s.io/kubernetes/pkg/labels"
+
+	"github.com/openshift/origin/pkg/client"
+)
+
+func NewClusterRoleReaper(roleClient client.ClusterRolesInterface, bindingClient client.ClusterRoleBindingsInterface) kubectl.Reaper {
+	return &ClusterRoleReaper{
+		roleClient:    roleClient,
+		bindingClient: bindingClient,
+	}
+}
+
+type ClusterRoleReaper struct {
+	roleClient    client.ClusterRolesInterface
+	bindingClient client.ClusterRoleBindingsInterface
+}
+
+// Stop on a reaper is actually used for deletion.  In this case, we'll delete referencing clusterrolebindings
+// then delete the clusterrole.
+func (r *ClusterRoleReaper) Stop(namespace, name string, timeout time.Duration, gracePeriod *kapi.DeleteOptions) (string, error) {
+	bindings, err := r.bindingClient.ClusterRoleBindings().List(labels.Everything(), fields.Everything())
+	if err != nil {
+		return "", err
+	}
+
+	for _, binding := range bindings.Items {
+		if binding.RoleRef.Name == name {
+			if err := r.bindingClient.ClusterRoleBindings().Delete(binding.Name); err != nil && !kerrors.IsNotFound(err) {
+				glog.Infof("Cannot delete clusterrolebinding/%s: %v", binding.Name, err)
+			}
+		}
+	}
+
+	// TODO decide about scanning through and deleting rolebindings that referenced the cluster role.  This isn't hard to do
+	// but on large clusters it could have big impacts
+
+	if err := r.roleClient.ClusterRoles().Delete(name); err != nil && !kerrors.IsNotFound(err) {
+		return "", err
+	}
+
+	return "", nil
```

Comment:
Was about to write, don't ask me why ;-) , because I have no idea. Actually I was thinking about proposing changing this method upstream just to return error.